File: algorithm/Python3_version/leetcode1276.number-of-burgers-with-no-waste-of-ingredients.py
class Solution:
    def numOfBurgers(self, tomatoSlices: int, cheeseSlices: int) -> List[int]:
        a = tomatoSlices - 2*cheeseSlices
        if a < 0 or a%2==1:
            return []
        juwuba = a//2
        xiaohuangbao = cheeseSlices - juwuba
        if xiaohuangbao >= 0:
            return [juwuba, xiaohuangbao]
        return []

File: algorithm/leetcode1276.number-of-burgers-with-no-waste-of-ingredients.cpp
class Solution {
public:
    vector<int> numOfBurgers(int tomatoSlices, int cheeseSlices) {
        int a = tomatoSlices - 2*cheeseSlices;
        if(a < 0 || (a&1)) return {};
        int juwuba = a/2;
        int xiaohuangbao = cheeseSlices - juwuba;
        if(xiaohuangbao >= 0)
            return {juwuba, xiaohuangbao};
        return {};
    }
};